### DISCUSSION TITLE
Are the properties `zotero-key` and `zt-attachments` mandatory?

### DISCUSSION THREAD

**Discussion by dcgk** [aidenlx/obsidian-zotero#387] [Help]
Hello!
I am excited to discover Zotlit! After years of being frustrated with the limitations of Nunjucks templating with Zotero Integration, I am here hoping to create a smoother workflow between Zotero and Obsidian.

Starting with the zt-field.eta  I was going to set up my properties, but soon discovered that there are two properties that I am not abel to edit: `zotero-key` and `zt-attachments`
Are these properties mandatory? Is there any way to change their name?

I keep my YAML organised using the pluggin Metadata Menu, does anybody have experience how to itegarte a Metadata Menu fileCalss into the zotlit template zt-field.eta? 

Much appreciated and many regards from Spain!

**Comment by aidenlx** [comment]
Half of this is gone in v2, and the other half opened up.

`zt-attachments` no longer exists. Attachments are read from `zt.attachments` in the template where you need them, instead of being parked in frontmatter.

`zotero-key` stays and cannot be renamed — it is how ZotLit identifies which Zotero item a note belongs to, and the vault-wide note index is built from it. `citekey` is the same kind of system field. Those two are the whole list.

Everything else is yours. Frontmatter fields are configured at **Settings > Templates > Frontmatter** as key / expression / language / merge strategy, so you can name properties to match a Metadata Menu fileClass. Keys ZotLit does not manage are preserved when a note is updated, and per-field merge strategies (**Replace** / **Append arrays** / **Keep existing**) let you protect values you edit by hand.

Reference: https://zotlit.aidenlx.site/docs/reference/templates/frontmatter

v2 beta is available now: https://zotlit.aidenlx.site/docs/install-zotlit
Migration guide: https://zotlit.aidenlx.site/docs/how-to/migrate-from-v1
